aws/sdk/integration-tests/s3/tests/alternative-async-runtime.rs

```diff
@@ -130,6 +130,7 @@ async fn retry_test(sleep_impl: Arc<dyn AsyncSleep>) -> Result<(), Box<dyn std::
         .credentials_provider(aws_types::credentials::SharedCredentialsProvider::new(
             credentials,
         ))
+        .retry_config(RetryConfig::standard())
```

Comment:
need to figure out how this plays with `aws-smithy-client::retry::Standard.config`

Reply:
Yeah, that's a good callout. I think we need to remove the retry policy from the client's generic args, since it really should be configured by the `RetryConfig` that feeds into `Config` rather than by the client builder. That way, the `RetryMode` in `RetryConfig` would set the policy instead.

If we want to retain the ability to customize the retry policy implementation completely for non-SDK Smithy clients, we could potentially add a `Custom` variant to `RetryMode`, but I don't think we want that for the SDK.

What are your thoughts on this?

Reply:
From discussion with @rcoh, it seems like removing the retry generic may be the best long term approach for simplifying this, but isn't currently possible due to `tower::retry::Policy` not being `dyn`-safe yet. This potential simplification should be punted to an RFC.